src/content/docs/kubernetes-pixie/kubernetes-integration/advanced-configuration/k8s-version2/troubleshooting.mdx

```diff
@@ -0,0 +1,294 @@
+    * `kube-state-metrics` is deployed using a custom deployment.
+    * There are multiple versions of `kube-state-metrics` running and the Kubernetes integration is not finding the correct one.
+
+    The Kubernetes integration automatically discovers `kube-state-metrics` in your cluster using this logic:
+
+    1. It looks for a `kube-state-metrics` service running on the `kube-system` namespace.
+    2. If that is not found, it looks for a service tagged with label `"k8s-app: kube-state-metrics"`.
+
+    The integration also requires the kube-state-metrics pod to have the label `k8s-app: kube-state-metrics` or `app: kube-state-metrics`. If neither of those are found, there will be a log entry like the following:
+
+    ```shell
+    2018-04-11T09:25:00.825532798Z time="2018-04-11T09:25:00Z" level=error 
+    msg="executing data source" data prefix="integration/com.newrelic.kubernetes" 
+    error="exit status 1" plugin name=nri-kubernetes stderr="time=\"2018-04-11T09:25:00Z\" 
+    level=fatal msg=\"failed to discover nodeIP with kube-state-metrics, 
+    got error: no pod found by label k8s-app=kube-state-metrics\"\n
+    ```
+
+    To solve this issue, add the `k8s-app=kube-state-metrics` label to the `kube-state-metrics` pod.
+  </Collapser>
+
+  <Collapser
+    id="metrics-missing"
+    title="Missing metrics for Namespaces, Deployments, and ReplicaSets"
+  >
+    If metrics for Kubernetes nodes, pods, and containers are showing but metrics for namespaces, deployments and `ReplicaSets` are missing, the Kubernetes integration is not able to connect to `kube-state-metrics`.
+
+    <CollapserGroup>
+      <Collapser
+        id="indicators"
+        title="Indicators of missing data"
+      >
+        Indicators of missing namespace, deployment, and `ReplicaSet` data:
+
+        * In the <DNT>**# of K8s objects**</DNT> chart, that data is missing.
+        * Queries for `K8sNamespaceSample`, `K8sDeploymentSample`, and `K8sReplicasetSample` don't show any data.
+      </Collapser>
+    </CollapserGroup>
+
+    There are few possible reasons for this:
+
+    1. `kube-state-metrics` service has been customized to listen on port 80. If that is the case, you may see an error like the following in the `verbose` logs:
+
+       ```shell
+       time="2018-04-04T09:35:47Z" level=error msg="executing data source" 
+       data prefix="integration/com.newrelic.kubernetes" error="exit status 1" 
+       plugin name=nri-kubernetes stderr="time=\"2018-04-04T09:35:47Z\" 
+       level=fatal msg=\"Non-recoverable error group: error querying KSM. 
+       Get http://kube-state-metrics.kube-system.svc.cluster.local:0/metrics: 
+       net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)\"\n"
+       ```
+
+       This is a known problem that happens in some clusters where it takes too much time for `kube-state-metrics` to collect all the cluster's information before sending to the integration.
```

Comment:
```suggestion
       This is a known problem, where some clusters where it takes too much time for `kube-state-metrics` to collect all the cluster's information before sending to the integration.
```

Reply:
I've rewritten this one to avoid the repetition of `where`. How about this?
`This is a known issue where some clusters take too much time for `kube-state-metrics` to collect all the cluster information before sending it to the integration.`